src/core_functions/scalar/list/list_has_any.cpp

```diff
@@ -0,0 +1,106 @@
+#include "duckdb/core_functions/lambda_functions.hpp"
+#include "duckdb/core_functions/scalar/list_functions.hpp"
+#include "duckdb/core_functions/create_sort_key.hpp"
+#include "duckdb/planner/expression/bound_cast_expression.hpp"
+#include "duckdb/common/string_map_set.hpp"
+
+namespace duckdb {
+
+static unique_ptr<FunctionData> ListHasAnyBind(ClientContext &context, ScalarFunction &bound_function,
+                                               vector<unique_ptr<Expression>> &arguments) {
+
+	arguments[0] = BoundCastExpression::AddArrayCastToList(context, std::move(arguments[0]));
+	arguments[1] = BoundCastExpression::AddArrayCastToList(context, std::move(arguments[1]));
+
+	const auto left_type = ListType::GetChildType(arguments[0]->return_type);
+	const auto right_type = ListType::GetChildType(arguments[1]->return_type);
+
+	if (left_type != LogicalType::SQLNULL && right_type != LogicalType::SQLNULL && left_type != right_type) {
+		LogicalType common_type;
+		if (LogicalType::TryGetMaxLogicalType(context, arguments[0]->return_type, arguments[1]->return_type,
+		                                      common_type)) {
+			arguments[0] = BoundCastExpression::AddCastToType(context, std::move(arguments[0]), common_type);
+			arguments[1] = BoundCastExpression::AddCastToType(context, std::move(arguments[1]), common_type);
+		} else {
+			throw BinderException("ListHasAny: cannot compare lists of different types: '%s' and '%s'",
+			                      left_type.ToString(), right_type.ToString());
+		}
+	}
+
+	return nullptr;
+}
+
+static void ListHasAnyFunction(DataChunk &args, ExpressionState &, Vector &result) {
+
+	auto &l_vec = args.data[0];
+	auto &r_vec = args.data[1];
+
+	if (ListType::GetChildType(l_vec.GetType()) == LogicalType::SQLNULL ||
+	    ListType::GetChildType(r_vec.GetType()) == LogicalType::SQLNULL) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		ConstantVector::GetData<bool>(result)[0] = false;
+		return;
+	}
+
+	const auto l_size = ListVector::GetListSize(l_vec);
+	const auto r_size = ListVector::GetListSize(r_vec);
+
+	Vector l_sortkey_vec(LogicalType::BLOB, l_size);
+	Vector r_sortkey_vec(LogicalType::BLOB, r_size);
+
+	const OrderModifiers order_modifiers(OrderType::ASCENDING, OrderByNullType::NULLS_LAST);
+	CreateSortKeyHelpers::CreateSortKey(ListVector::GetEntry(l_vec), l_size, order_modifiers, l_sortkey_vec);
+	CreateSortKeyHelpers::CreateSortKey(ListVector::GetEntry(r_vec), r_size, order_modifiers, r_sortkey_vec);
```

Comment:
If `l_vec` and/or `r_vec` are dictionary vectors, are the resulting `l_sortkey_vec` and `r_sortkey_vec` also dictionary vectors? I am wondering because we assume they are flat later on.

Reply:
No, the sortkey vector are created as flat vectors and only changed to constant vectors if the input is constant as well. And as far as I know, list child vectors are always flat?